slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/InterchangeImportConflicts.java

```diff
@@ -0,0 +1,159 @@
+
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlock;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.common.io.Resources;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.junit.jupiter.api.Test;
+
+public class InterchangeImportConflicts extends InterchangeBaseIntegrationTest {
+
+  @Test
+  void duplicateEntriesAreNotInsertedToDatabase() throws IOException {
+    final URL importFile = Resources.getResource("interchange/singleValidBlock.json");
+    slashingProtection.importData(importFile.openStream());
+    slashingProtection.importData(importFile.openStream()); // attempt to reimport
+    jdbi.useHandle(
+        handle -> {
+          final List<SignedBlock> blocksInDb = findAllBlocks(handle);
+          assertThat(blocksInDb).hasSize(1);
+          assertThat(blocksInDb.get(0).getSlot()).isEqualTo(UInt64.valueOf(12345));
+          assertThat(blocksInDb.get(0).getValidatorId()).isEqualTo(1);
+          assertThat(blocksInDb.get(0).getSigningRoot())
+              .isEqualTo(
+                  Optional.of(
+                      Bytes.fromHexString(
+                          "0x4ff6f743a43f3b4f95350831aeaf0a122a1a392922c45d804280284a69eb850b")));
+        });
+  }
+
+  @Test
+  void canloadConflictingBlocksInSameSlot() throws IOException {
+    final URL importFile = Resources.getResource("interchange/conflictingBlocks.json");
+    slashingProtection.importData(importFile.openStream());
+    jdbi.useHandle(
+        handle -> {
+          final List<SignedBlock> blocksInDb = findAllBlocks(handle);
+          assertThat(blocksInDb).hasSize(2);
+          assertThat(blocksInDb.get(0).getSlot()).isEqualTo(UInt64.valueOf(12345));
+          assertThat(blocksInDb.get(0).getValidatorId()).isEqualTo(1);
+          assertThat(blocksInDb.get(0).getSigningRoot())
+              .isEqualTo(
+                  Optional.of(
+                      Bytes.fromHexString(
+                          "0x4ff6f743a43f3b4f95350831aeaf0a122a1a392922c45d804280284a69eb850b")));
+
+          assertThat(blocksInDb.get(1).getSlot()).isEqualTo(UInt64.valueOf(12345));
+          assertThat(blocksInDb.get(1).getValidatorId()).isEqualTo(1);
+          assertThat(blocksInDb.get(1).getSigningRoot())
+              .isEqualTo(
+                  Optional.of(
+                      Bytes.fromHexString(
+                          "0x4ff6f743a43f3b4f95350831aeaf0a122a1a392922c45d804280284a69eb850c")));
+        });
+  }
+
+  @Test
+  void canLoadFileWithDuplicateBlocks() throws IOException {
+    final URL importFile = Resources.getResource("interchange/duplicateBlocks.json");
+    slashingProtection.importData(importFile.openStream());
+    slashingProtection.importData(importFile.openStream()); // attempt to reimport
```

Comment:
What is the difference between this test and the duplicateEntriesAreNotInsertedToDatabase test? Both are inserting the same block multiple times I think and check there is only a single entry.